### ISSUE TITLE
Suggestion: better document how to toggle ignore mode

### ISSUE THREAD

**Issue by indeedwatson** [tridactyl/tridactyl#2108]
I just spent about 20 minutes reading the help pages and trying out different
commands to add a bind to enter or toggle ignore mode, only to realize that one
of the listed keybinds does that. It should be more explicit that Shift + Escape
will do this, when looking through the docs with keyboards such as `disable` the
most prominent section is the url blacklisting, which is not what one would want
when one is dealing for the first time with a page where you need to disable
tridactyl.

In fact I'd say this toggle is one of the most important things a person new to
this addon should learn, as it could become frustating if it gets in your way
and you don't know how to disable it.

**Comment by glacambre** [comment]
`disable` is the third result when you search for it in the help pages, I'm don't know how we could make this easier to find. Did you try other keywords? We could add them to `:mode ignore`'s documentation.

**Comment by indeedwatson** [comment]
I think adding `you can toggle ignore mode with <S-Esc>` would be helpful.

**Comment by bovine3dom** [comment]
It's currently on the first page of the tutorial such appears when you first install the extension. Is there anything we could have done to make it more digestible?

**Comment by indeedwatson** [comment]
> It's currently on the first page of the tutorial which appears when you first install the extension. Is there anything we could have done to make it more digestible?

I must've missed it the first time, and didn't know it was there later (nor did I know how to access that page again). My instinct was to go to the help page and search for `disable`, which led me to eventually find how to do it via the command, and later I actually mapped it to `<C-Esc>` before finding that it was already a thing with S-Esc.

So it's up to you guys, it might have just been me having a brain fart for 20 minutes, in hindsight I see that what I was looking for comes up in the `ignoremaps` part but from my perspective it's not clear that those binds will take you *into* ignore mode. It makes it sound like they'll take you out of the mode, and that to enter it you must type `mode ignore` as a command. See:

> Make sure you're aware of the key bindings (ignoremaps) that will allow you to go come back to normal mode from ignore mode before you run :mode ignore otherwise you're going to have a hard time re-enabling Tridactyl.

Feel free to close this if you deem the wording is fine as it is.

**Comment by bovine3dom** [comment]
We do want to improve the wording :)

It isn't supposed to take 20 minutes to work out how to enter ignore mode.

`mode`'s help page will already list the S-Esc binding. It currently displays the exact bind only when you hover over it. I suspect the amount of luck one needs to find that out is too high.

**Comment by glacambre** [comment]
>I suspect the amount of luck one needs to find that out is too high

I agree (in fact that's what I've been thinking ever since I implemented that) but I haven't been able to come up with a solution that would convey that more information is available without cluttering up the help pages.

**Comment by ntc2** [comment]
I also just had trouble with this, looking at the `:help` docs. My goal was to bind `<Insert>` to toggle between normal mode and ignore mode. The solution I eventually came up with was this:

    bind --mode=normal <Insert> mode ignore
    bind --mode=ignore <Insert> mode normal

Would have been helpful to me for the docs on `ignore` to explain how to do this.

**Comment by bovine3dom** [comment]
Thanks for mentioning it, ntc2. I'm not sure it makes much sense to add a reference to `:bind` on `:mode` given that the same argument could apply to all ex-commands - `:bind` is kind of related to all of them. But maybe we could display it more prominently somewhere else. It's already on the new tab page. Perhaps it just needs moving up a little bit, or rewording?

**Comment by ntc2** [comment]
> Thanks for mentioning it, ntc2. I'm not sure it makes much sense to add a reference to `:bind` on `:mode` given that the same argument could apply to all ex-commands - `:bind` is kind of related to all of them. But maybe we could display it more prominently somewhere else. It's already on the new tab page. Perhaps it just needs moving up a little bit, or rewording?

Not sure ... maybe do nothing? Hopefully anyone with the same question will eventually find this issue. Part of the difficulty for me was using `bind` with `--mode`: I already had some `bind`s in my config with no mode argument, and so a conceptual leap was needed to get to the version I needed here.

> > (For anyone finding this via google, you can reload your config with :source after making changes).
> 
> FWIW, this only applies to RC files. If you run `:bind` in the browser it will take effect immediately and persist.

Where would this persist? Would it write the new binds to my config file? This is counter intuitive.

**Comment by bovine3dom** [comment]
I see what you mean. We could mention `:bind --mode` on the `:mode` help. We could also make sure we explicitly mention the mode in our examples everywhere so you never see `:bind k` but always `:bind --mode=normal k`.

> Where would this persist? Would it write the new binds to my config file? This is counter intuitive.

It is extremely counter-intuitive, I agree :)

The main issue with your reasoning is that an RC file is not a config file. RC stands for "run commands", which is literally what it does: it is a list of commands that Tridactyl runs at startup and when you run `:source`. You can fill it full of `tabopen` or `winclose` if you want.

Some commands in Tridactyl change Tridactyl's configuration (`:set` and `:bind` for example). That configuration is stored in Firefox's internal storage, which is a SQLite database. Tridactyl includes a command, `:mkt`, which parses that stored configuration, converts it into a list of commands that would generate that configuration and then stores those commands as an RC file.

It's not a million miles away from what Vim does: if you change a setting in Vim, that change gets stored in memory and it isn't written to your RC file, because your RC file isn't your configuration. Your configuration is stored in memory. The difference with Tridactyl is that the configuration is stored on disk in the Firefox internal storage because that makes development easier, and usage easier as end-users don't have to install the native messenger.

But to reiterate: I agree with you, it's confusing.
